lib/DIKUrevy/Website/Controller/Auth.pm

```diff
@@ -84,6 +88,77 @@ sub create_user_submit {
     return $self->redirect_to('frontpage');
 }
 
+sub random_password() {
+    my @chars = ('A'..'Z', 'a'..'z', '0'..'9', '@%$#&+*');
+    return join('', map { $chars[rand @chars] } 1..16);
+}
+
+sub new_password {
+    my $self = shift;
+
+    my $jwt = $self->req->query_params->param('jwt');
+    if ($jwt) {
```

Comment:
Jeg ville nok smide JWT-endpointet i sin egen rute, for at holde ruterne simple. :)